### PR TITLE
CORTX-31950: add url to kwargs for S3Client constructor

### PR DESCRIPTION
**Pull request by avrnv**
# Problem Statement
- [CORTX-31950](https://jts.seagate.com/browse/CORTX-31950): CSM-S3 communication should be on https.

# Design
-  Per Pranay's request make S3 client capable of handling URL provided as is instead of constructing from host, port and tls_enabled arguments.

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing
- Are test cases updated along with code changes due to Enhancements/Bugs [Y/N]:
- Confirm that new test cases are added to regression and sanity plan files and relevant feature plan files [Y/N]:
- Confirm that Test Cases are added for new features added [Y/N]: 
- Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- Confirm Testing was performed with installed RPM/K8s deployment [Y/N]:  

# Review Checklist 
####  Before posting the PR please ensure:
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] New package/s added to setup.py?
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified.

# KVstore/Confstore feature (changes/fixes) checklist
#### Confirm changes are made for:
- [ ] ConfStore
- [ ] KVStore
- [ ] ConfCli
- [ ] Test cases added for all above 3
- [ ] changes done in all the KVpayloads (ConsulKvPayload, IniKvPayload, KvPayload)

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

**Review by pawankumarsrivastava — APPROVED** [review]

File: py-utils/src/utils/s3/client.py [hunk 1]
```diff
@@ -34,7 +35,7 @@ class S3Client(HttpClient):
     def __init__(
         self, access_key_id: str, secret_access_key: str,
         host: str = 'localhost', port: int = 8000,
-        tls_enabled: bool = False, ca_bundle: str = '',
+        tls_enabled: bool = False, url: str = '', ca_bundle: str = '',
```
Comment (ujjwalpl) [inline py-utils/src/utils/s3/client.py:38]: Lets avoid redundancy in the parameters. Lets accept url instead of host and port.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/s3/client.py (hunk 1 above)
Comment (avrnv) [reply py-utils/src/utils/s3/client.py]: Done.

**Review by ujjwalpl — APPROVED** [review]

File: py-utils/src/utils/s3/client.py [hunk 2]
```diff
@@ -45,11 +45,15 @@ def __init__(
         :param host: hostname of the S3 server.
         :param port: port of the S3 server.
         :param tls_enabled: flag to use https.
+        :param url: whole URL for the client, if set, then host, port and tls_enabled are ignored.
```
Comment (cdeshmukh) [inline py-utils/src/utils/s3/client.py:48]: > then host, port and tls_enabled are ignored.

these arguments are removed, nothing there to ignore from the list. this comment needs update

Comment (abhijit1patil) [reply py-utils/src/utils/s3/client.py:48]: Handled the comment in New PR: https://github.com/Seagate/cortx-utils/pull/833

**Comment by abhijit1patil** [comment]
## Note: To handle the review comments and to update PR description, we created new PR : https://github.com/Seagate/cortx-utils/pull/833. 
We will close this PR, once other New PR is approved by peers

File: py-utils/src/utils/s3/client.py (hunk 2 above)
Comment (cdeshmukh) [reply py-utils/src/utils/s3/client.py:48]: why not here? review comments are supposed to be addressed in same PR
Also 833 is in draft state, reviewers there would be confused what is new in that PR, given same code here was approved?
@pawankumarsrivastava can you help team align with process

Comment (abhijit1patil) [reply py-utils/src/utils/s3/client.py:48]: @cdeshmukh 
As you can see this is Alex's PR and alex is not available and I don't have rights to update this PR. It is not possible for me to update PR description and handle your review comments. That's why I clone same branch in my repo and raised the PR.
https://github.com/Seagate/cortx-utils/pull/833 is review state now. 
If you think this not correct way to handle this scenario, please suggest alternative way
CC: @pawankumarsrivastava

Comment (cdeshmukh) [reply py-utils/src/utils/s3/client.py:48]: No didnt realize that you are not original author, now i understand better. So review comments can be handled separately then.

Also i still dont see the message about 1N deployment expected failure, i had asked to update PR with same, how some one supposed to know if anyone looking just at PR.

And please use full name 'Alexander' as Alex is confusing with Alex scan.

Comment (abhijit1patil) [reply py-utils/src/utils/s3/client.py:48]: Regarding 1N deployment expected failure: 
I thought you meant adding to PR description. I added those things in New PR.. I will update everything here as well.

**Comment by abhijit1patil** [comment]
## NOTE: 
This PR depends on https://github.com/Seagate/cortx-manager/pull/852, because of interdependency with Cortx-manager PR, One node deployment is failing
I will create the custom build using both utils and CSM PR. I will update the build and deployment status here.
@cdeshmukh @sachinpunadikar @ujjwalpl @pawankumarsrivastava

**Comment by cdeshmukh** [comment]
> [abhijit1patil](https://github.com/abhijit1patil) [9 minutes ago](https://github.com/Seagate/cortx-utils/pull/824#discussion_r895330126)
Regarding 1N deployment expected failure:

@abhijit1patil put reason/s why its expected. 

And lets merge this original PR itself then, you address review comments only in new PR

**Comment by cdeshmukh** [comment]
> This PR depends on https://github.com/Seagate/cortx-manager/pull/852, because of interdependency with Cortx-manager PR, One node deployment is failing


Add CSM PR link here

> I will create the custom build using both utils and CSM PR. I will update the build and deployment status here.

Yes do this here

**Comment by abhijit1patil** [comment]
@cdeshmukh 
CSM PR link : https://github.com/Seagate/cortx-manager/pull/852

**Comment by abhijit1patil** [comment]
@cdeshmukh 

## Custom Build with Utils and CSM PR
- [Custom Build Job](https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6649/)
- Custom Build for deployment
cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-6649-custom-ci
cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-6649-custom-ci
cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-6649-custom-ci

- [Deployment Job](https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/6653/)
- 1Node deployed VM
image

**Comment by cdeshmukh** [comment]
> @cdeshmukh CSM PR link : [Seagate/cortx-manager#852](https://github.com/Seagate/cortx-manager/pull/852)

oh it was same which was mentioned already, got confused with wordings

**Review by cdeshmukh — APPROVED** [review]

**Pull request merged**